basics/devtools.md

```diff
@@ -0,0 +1,145 @@
+   the “Sources” tab and in the sidebar expand `vivaldi-data` and then `css-mods`.
+   Click on your mod file, the contents will be shown right next to it. Since we
+   know how Vivaldi changes the color of the active tab, we can use the same
+   code in our file to overwrite Vivaldi’s instructions. Change the color to
+   `green` this time and notice how the change is immediately visible.
+
+   ![edit file image]
+
+7. This is a temporary modification too. However, we can make our edit permanent
+   by right‐clicking and selecting “Save as...”. Choose the location of the mod
+   file to overwrite, or save the edit as backup.
+
+### Reloading the user interface
+
+The standard way of performing edits is opening our mod file in a proper text
+editor, editing, saving and reloading the UI. Here a few options to load the
+changes:
+
+- Exit Vivaldi and open it again.
+- Open a new browser window.
+- Select all tabs and move them to a new window. This is available as a menu
+  entry when right‐clicking a tab and has the advantage of keeping webpages and
+  media playback active.
+- Restart Vivaldi. The URL **`vivaldi://restart/`** will restart Vivaldi when it
+  is being opened in the current tab. We can either enter the address directly
+  in the address field, or use a bookmark. One nifty option is to save the
+  address, assign a nickname to it (e.g. `re`) and load it with quick commands.
+  `shift‐/alt-enter` will open in current tab, when bookmarks are set to open in
+  a new tab by default and vice versa; this is dependent on our search,
+  bookmarks and quick commands settings.
+
+All the actions load the `browser.html` file anew, which in turn loads our
+modification files.
+
+### Console
+
+Let’s write a “Hello, World!” example script for testing our Javascript setup as
+described in [Installation][installation link].
+
+1. Open the Javascript modification file in a text editor.
+2. Copy and paste following code and save.
+
+   ```javascript
+   setTimeout(function wait() {
+     const browser = document.getElementById("browser");
+     if (browser) console.log("Hello, World!");
+     else setTimeout(wait, 300);
+   }, 300);
+   ```
+
+3. Reload the user interface and load DevTools.
+4. Click the “Console” tab, our message should be visible.
+
+   ![hello world image]
+
+5. Now we introduce an error. Delete the first line of the script and reload UI.
+6. Open DevTools and switch to the console. Vivaldi detects the error at line 4,
+   where the closing bracket is located. We deleted the opening bracket, so this
+   is expected. Console logs are one way to test the functionality of our mods
+   and find errors—use them to your advantage.
+
+   ![error image]
+
+---
+
+I hope these instructions have been helpful. For further questions post a reply
+to the [original topic][topic link]; we have a healthy community willing to help
+you out.
+
+[devtools image]: /assets/basics/devtools/devtools.png
```

Comment:
Should these use a relative pattern instead? Am I missing something?

Reply:
Both have their disadvantages. An absolute link works only if the parser is smart enough, a relative link breaks when you move the file where it is used (but both links break if you move the targets).